pandas/_libs/tslib.pyx

```diff
@@ -455,18 +455,18 @@ cpdef array_to_datetime(
         set out_tzoffset_vals = set()
         tzinfo tz_out = None
         bint found_tz = False, found_naive = False
-        cnp.broadcast mi
+        cnp.flatiter it = cnp.PyArray_IterNew(values)
 
     # specify error conditions
     assert is_raise or is_ignore or is_coerce
 
     result = np.empty((<object>values).shape, dtype="M8[ns]")
-    mi = cnp.PyArray_MultiIterNew2(result, values)
     iresult = result.view("i8").ravel()
 
     for i in range(n):
         # Analogous to `val = values[i]`
-        val = <object>(<PyObject**>cnp.PyArray_MultiIter_DATA(mi, 1))[0]
+        val = cnp.PyArray_GETITEM(values, cnp.PyArray_ITER_DATA(it))
```

Comment:
elsewhere we use `val = (<int64_t*>cnp.PyArray_ITER_DATA(it))[0]`.  should we use PyArray_GETITEM there too?  seems cleaner

Reply:
This returns a PyObject I think, so I don't think we can use it unless the array is object dtype.